vyper/context/validation/local.py

```diff
@@ -350,9 +369,20 @@ def visit_Expr(self, node):
             raise StructureException("To call an event you must use the `log` statement", node)
 
         if isinstance(fn_type, ContractFunctionType):
-            if self.func.is_constant and not fn_type.is_constant:
+            if self.func.mutability is StateMutability.VIEW and fn_type.mutability not in (
+                StateMutability.VIEW,
+                StateMutability.PURE,
+            ):
+                raise ConstancyViolation(
+                    "Cannot call a mutating function from a view function", node
```

Comment:
the word "mutating" feels a bit out of place here.  it's technically correct, but i think most users would better understand "state modifying" or something similar.

Reply:
```suggestion
                    "Cannot call a state modifying function from a view function", node
```

Reply:
Yess